src/dvc_data/hashfile/meta.py

```diff
@@ -64,8 +66,14 @@ def from_info(
 
     @classmethod
     def from_dict(cls, d: Dict[str, Any]) -> "Meta":
-        kwargs = {field: d[field] for field in fields_dict(cls) if field in d}
+        kwargs = {}
+        for field in cls.fields:
+            if field in d:
+                kwargs[field] = d[field]
         return cls(**kwargs)
 
     def to_dict(self) -> Dict[str, Any]:
         return asdict(self, recurse=False, filter=_filter_default_or_none)
+
+
+Meta.fields = list(fields_dict(Meta))
```

Comment:
@efiop, let me know what you think of this (if your issue is with adding class variable):
```suggestion
META_FIELDS = list(fields_dict(Meta))
```

It is 3-4x slow if we call `fields_dict()` inside a hot loop, so I'd like to avoid that.

Reply:
Maybe I'm not undersatnding the tradeoff here. Just looks hacky 😅 But not that bad. I have no major problems with this.